assets/scripts/streets/Geolocation.jsx

```diff
@@ -0,0 +1,156 @@
+
+const zoomLevel = 12
+
+class Geolocation extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      mapCenter: [40.645, -73.975]
+    }
+
+    this.searchResults = this.searchResults.bind(this)
+    this.markerDrag = this.markerDrag.bind(this)
+    this.gotClick = this.gotClick.bind(this)
+    this.hidePopup = this.hidePopup.bind(this)
+  }
+
+/* start click event function */
+  gotClick (e) {
+    const displayAddressData = (res) => {
+      setSettings({ addressInformationLabel: res.features[0].properties.label })
+      setSettings({ markerLocation: res.features[0].geometry.coordinates.reverse() })
+
+      this.map.leafletElement.panTo(this.props.markerLocation)
+      console.log('Click Point Results : ', this.props.markerLocation, this.props.addressInformationLabel)
+    }
+
+    function gotAddressData (res, err) {
+      res.json().then(displayAddressData)
+    }
+
+    const options = {
+      lat: e.latlng.lat,
+      lng: e.latlng.lng
+    }
+
+    const clickUrl = `${apiurlReverse}?api_key=${apikey}&point.lat=${options.lat}&point.lon=${options.lng}`
+    fetch(clickUrl).then(gotAddressData)
+  }
+
+/* end click event function */
+
+/* start on marker drag function */
+  markerDrag (e) {
+    const gotJson = (res) => {
+      this.setState({
+        renderPopup: true
+      })
+
+      setSettings({ addressInformationLabel: res.features[0].properties.label })
+      setSettings({ markerLocation: e.target.getLatLng() })
+      console.log('Drag End Results : ', e.target.getLatLng(), this.props.addressInformationLabel)
+    }
+
+    function gotAddress (res, err) {
+      res.json().then(gotJson)
+    }
+
+    const targetCoords = e.target.getLatLng()
+
+    const dragEndUrl = `${apiurlReverse}?api_key=${apikey}&point.lat=${targetCoords.lat}&point.lon=${targetCoords.lng}`
+    fetch(dragEndUrl).then(gotAddress)
+  }
+
+/* end on marker drag function */
+
+// p = point , l = label
+  searchResults (p, l) {
+    this.setState({
+      addressName: l,
+      mapCenter: p,
+      markerLocation: p
+    })
+
+    this.map.leafletElement.panTo(this.props.markerLocation)
+  }
+
+  hidePopup (e) {
+    this.setState({
+      renderPopup: false
+    })
+  }
+
+  render () {
+    const markers = this.props.markerLocation ? (
+      <Marker
+        position={this.props.markerLocation}
+        onDragEnd={this.markerDrag}
+        onDragStart={this.hidePopup}
+        draggable
+      />
+    ) : null
+
+    let popup = this.props.markerLocation ? (
+      <Popup
+        position={this.props.markerLocation}
+        maxWidth={300}
+        closeOnClick={false}
+        closeButton={false}
+        offset={[0, -25]}
+      >
+        <span>{this.props.addressInformationLabel}</span>
+      </Popup>
+    ) : null
+
+    if (this.state.renderPopup === false) {
+      popup = null
+    }
+
+    return (
+      <div id='rootDiv'>
+        <div id='dark-border'>
```

Comment:
Similarly, let's use classnames instead of `id` here, and in the line below.

What we like to do is namespace classnames so that there's less chance of class names colliding with other class names in other parts of the application. I would suggest using a class name like `.geolocation-border` and `.geolocation-input`.